Debezium Platform: Mid-Year Update (2026) (#1284)
debezium/debezium.github.io: _posts/2026-07-01-debezium-platform-status-update.adoc

---
layout: post
title:  "Debezium Platform: Mid-Year Update (2026)"
date:   2026-07-01
tags: [debezium-platform, news, update, ui, features]
featured: true
author: mfvitale
---

Hello Debezium community, Mario here.
I have been leading the development of the Debezium Platform for nearly two years now, and it has been about a year since my last update on the project.
I took a moment to reflect on what we have worked on recently, what we intend to work on by the end of the year, and the current status of the Debezium Platform.

+++<!-- more -->+++

A lot of work was done in the last year, but I don't want to mimic a release note, even though I greatly value small fixes and features.
Instead, I would like to highlight what I think, in my opinion, is really worth mentioning.

== What's new

=== Enhanced configuration UI

As you know, Debezium has a lot of configurations that give the flexibility to adapt and fine-tune it to different users' needs.

> Some configurations are cross-connectors, while others are connector-specific, and others are just related to a very specific functionality.

Reviewer: ```suggestion
Some configurations are cross-connector, while others are connector-specific, and others are just related to a very specific functionality.
```

> Debezium users, including both Kafka Connect connectors and the server, are used to work with text-based configuration files.

Reviewer: ```suggestion
Debezium users, including those working with both Kafka Connect connectors and the server, are accustomed to working with text-based configuration files.
```

So they discover the configurations and their meaning in the documentation, and that's fine.

> The idea behind the Debezium Platform was, and still is, to offer users a simplified way to define data pipelines, focusing on its design.

Reviewer: ```suggestion

The idea behind the Debezium Platform was, and still is, to offer users a simplified way to define data pipelines, with a focus on design.
```

> From the beginning, we had it clear that users should easily configure each piece of the pipeline with a clear interface showing all the possible configurations, what is required, and eventually a default value.

Reviewer 2: Do you think we should also mention the conditional property fields

https://github.com/user-attachments/assets/99bbbb8d-d3e9-433e-b1d9-f1f549a50524

Author: WDYT of 

```text
From the beginning, we were clear that users should be able to configure each piece of the pipeline through a clear interface that presents all available options, indicates which settings are required, provides sensible default values where applicable, and supports conditional configuration based on other selected options.
```

Reviewer 2: yes 👍

> Nothing special, just something that a UI user should expect.

Reviewer: ```suggestion
Nothing special, just something that any user would expect in a user interface.
```

We worked on this in the first part of the year, so we moved from the following UX.

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-source-before.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

to this

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-source-new.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

> This is not intended to be a technical post, but I have to say that this was not simple, since this impacted different debezium components.

Reviewer: ```suggestion
This is not intended to be a technical post, but I have to say it was not simple, as it affected multiple Debezium components.
```

If you are curious about the details, you can have a look at https://github.com/debezium/debezium-design-documents/blob/main/DDD-13.md#ddd-13-debezium-component-descriptors[DDD-13: Debezium Component Descriptors].

There are still some things to improve.

> For example, configuration grouping and order could not be perfect, but at least now there is the ground to improve them.

Reviewer: ```suggestion
For example, configuration grouping and order are not perfect, but at least there is now a basis for improving them.
```

=== Integrated monitoring

> One of the most common requested features for the day 2 activities was the observability.

Reviewer: ```suggestion
One of the most commonly requested features for the day 2 activities was observability.
```

> Debezium largely adopted JMX as the way to expose metrics in the connector and server, leaving to the end users the observability stack https://debezium.io/documentation/reference/stable/operations/monitoring.html[setup and configuration].

Reviewer: ```suggestion
Debezium largely adopted JMX as the way to expose metrics in the connector and server, leaving end users with the observability stack https://debezium.io/documentation/reference/stable/operations/monitoring.html[setup and configuration].
```

> With the Platform, we wanted to simplify as much as possible the setup, providing an integrated dashboard to monitor your pipelines.

Reviewer: ```suggestion
With the Platform, we wanted to simplify setup as much as possible by providing an integrated dashboard to monitor your pipelines.
```

</div>
++++

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-monitoring-streaming-panels.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-monitoring-snapshot-panels.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-monitoring-snapshot-running-panels.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

++++
<div class="imageblock centered-image">
    <img src="/assets/images/2026-07-01-debezium-platform-status-update/debezium-platform-monitoring-toolbar.png" style="max-width:100%;" class="responsive-image" alt="Pipeline design">
</div>
++++

> We opted for the https://opentelemetry.io/[OpenTelemetry] standard, collecting metrics from Debezium Server with https://opentelemetry.io/docs/zero-code/java/agent/[otel java agent] via https://opentelemetry.io/docs/collector/[OpenTelemetry Collector].

Reviewer: ```suggestion
We opted for the https://opentelemetry.io/[OpenTelemetry] standard, collecting metrics from Debezium Server with https://opentelemetry.io/docs/zero-code/java/agent/[OpenTelemetry Java Agent] via https://opentelemetry.io/docs/collector/[OpenTelemetry Collector].
```

For those interested in the details, you can take a look at https://github.com/debezium/debezium-design-documents/blob/main/DDD-38.md#ddd-38-pipeline-monitoring-for-debezium-platform[DDD-38: Pipeline Monitoring for Debezium Platform] and https://debezium.io/documentation/reference/stable/operations/debezium-platform.html#_pipeline_monitoring[documentation].

=== Bonus: AI experiments

> At the end of last year, we did a POC about integrating an AI chatbot in the Platform with the idea of letting users interact with the Platform in a conversational way.

Reviewer: ```suggestion
At the end of last year, we designed a POC to integrate an AI chatbot into the Platform, with the idea of letting users interact with it in a conversational way.
```

We got some good results, but the status is not yet at a point where we want to release it.
We plan to continue working on it throughout the year. So keep an eye on it.

== What's next

=== Alerting

> Having a monitoring dashboard is good, but having a way to be proactively alerted when things start going bad is much better.

Reviewer: ```suggestion
Having a monitoring dashboard is good, but having a way to be proactively alerted when things start to go wrong is much better.
```

This is why we want to offer the possibility to configure alerts to be notified when a certain metric reaches a defined threshold.

=== Identity Access Management

> One of the biggest limitations of the Platform today is that it assumes a trusted environment.

Reviewer: ```suggestion
One of the biggest limitations of the Platform today is its assumption of a trusted environment.
```

This is perfectly fine when you are evaluating it locally or running it in a small team, but it quickly becomes an issue when multiple users and teams need to share the same instance.

This is why Identity and Access Management is one of the next major areas we want to tackle.

The goal is not just to add a login page.
We want to provide proper authentication and authorization so that administrators can control who can access the Platform and what they are allowed to do.
For example, some users may only need to monitor pipelines, while others should be able to create, modify, or delete them.
Likewise, in a shared environment, different teams should be able to manage only their own pipelines, without seeing or accidentally modifying pipelines owned by other teams, while administrators retain full visibility across the entire Platform.

> This will also provide the foundation for future enterprise-oriented features, making the Platform more suitable for shared environments and production deployments.

Reviewer: ```suggestion
This will also lay the foundation for future enterprise-oriented features, making the Platform better suited to shared environments and production deployments.
```

We are still defining the details, but the current direction is to integrate with standard identity providers, allowing organizations to leverage their existing authentication infrastructure instead of managing users separately.

=== Host-based deployment

> There was a recurring request from the community to have a Debezium Server's UI, even though we are offering the Debezium Platform.

Reviewer: ```suggestion
There was a recurring request from the community for a Debezium Server UI, even though we are offering the Debezium Platform.
```

> The reason is that Debezium Platform, though it was designed to deploy a pipeline in different environment, only supports Kubernetes as of now.

Reviewer: ```suggestion
The reason is that Debezium Platform, though it was designed to deploy pipelines across different environments, only supports Kubernetes as of now.
```

This is going to change.

https://github.com/d1vyanshu-kumar[Divyanshu Kumar], a https://summerofcode.withgoogle.com/[Google Summer Of Code] contributor, is working on https://github.com/debezium/debezium-design-documents/pull/45[DDD-41: Host-Based Pipeline Deployment for the Debezium Platform].

> With his work, the Platform will provision your bare metal machines via Ansible to prepare them to run your pipelines.

Reviewer: ```suggestion
With his work, the Platform will provision your bare-metal machines via Ansible, preparing them to run your pipelines.
```

This is not a user interface served directly by Debezium Server; it's much more.
You can manage multiple Debezium Server instances (pipelines) from the same dashboard.
https://github.com/debezium/dbz/issues/2085[DBZ-2085] tracks the feature progress.

== Conclusion

I hope this helped you to have an overview of the current Platform status and where we would like to go.

> We are working hard to bring to the community a complete Platform that lets you use all Debezium components in an easy way.

Reviewer: ```suggestion
We remain steadfast in making the Platform a complete solution that enables the community to easily use the entire Debezium ecosystem of components.
```
